ledger/store/src/block/mod.rs

```diff
@@ -392,6 +392,11 @@ pub trait BlockStorage<N: Network>: 'static + Clone + Send + Sync {
         self.transaction_store().finish_atomic()
     }
 
+    /// Either enables or disables the atomic override.
+    fn flip_atomic_override(&self) -> Result<bool> {
+        self.state_root_map().flip_atomic_override()
```

Comment:
Why are you only flipping one map?

Reply:
the map is just a means to access the storage; since the override is only useful and effective with persistent storage, it is implemented at the lowest level and any map can be used to toggle it

Reply:
I don't get it.

What is it about `state_root` and not anything else that impacts this?

`state_root` is not even touched by finalize scope.

Reply:
The state root map (the choice of which is insignificant, it's just the first map listed in `BlockStorage`), like all the other instances of `DataMap`, contains a reference to a single, shared instance of `RocksDB`, so any of them (and either from `BlockStore` or `FinalizeStore`, as they are the ones available when inserting blocks) can be used to toggle the override, which is implemented within the `RocksDB` object. Once the override is toggled (from any `DataMap`), all the other maps are aware of it being in force, and when they call `finish_atomic`, they only extend the serialized atomic batch, but it doesn't get executed (which would normally happen with the final/outermost call) until the override is flipped again.

Note that `MemoryMap` - which doesn't have a single underlying instance of storage - is irrelevant here (as only persistent storage needs atomic operations), and only has a noop impl of the override.